evaluations/src/evaluators/mod.rs

```diff
@@ -55,13 +61,26 @@ pub(crate) async fn evaluate_inference(
         send_feedback,
     } = params;
     let EvaluationConfig::Inference(inference_evaluation_config) = &*evaluation_config;
+
+    // Filter evaluators based on cancellation tokens (if provided)
+    let evaluators_to_run: Vec<_> = inference_evaluation_config
+        .evaluators
+        .keys()
+        .filter(|name| {
+            // If no cancellation tokens provided, run all evaluators
+            // Otherwise, only run evaluators whose tokens are not cancelled
+            cancellation_tokens
+                .is_none_or(|tokens| tokens.get(*name).is_none_or(|token| !token.is_cancelled()))
+        })
+        .collect();
```

Comment:
You can get rid of this `collect` call, as pass the iterator directly to `FuturesUnordered::from_iter`